File: examples/fibbench.js
function fib(n) {
    if (n <= 1) return n;
    return fib(n - 1) + fib(n - 2);
}

function main() {
    const iterations = 1000;
    
    // Warmup the JIT compiler slightly to get realistic 'warm' metrics
    for(let i=0; i<10; i++) fib(19);

    const start = process.hrtime.bigint();
    
    let result = 0;
    for (let i = 0; i < iterations; i++) {
        result += fib(19);
    }
    
    const end = process.hrtime.bigint();
    const totalMs = Number(end - start) / 1000000.0;
    const avgMs = totalMs / iterations;
    
    console.log("--- Node.js Benchmark ---");
    console.log(`Fibonacci(19) runs: ${iterations}`);
    console.log(`Total time: ${totalMs.toFixed(2)} ms`);
    console.log(`Time per execution: ${avgMs.toFixed(2)} ms`);
    console.log(`Sanity check result sum: ${result}`);
}

main();

File: examples/fibbench.py
import time

def fib(n):
    if n <= 1: return n
    return fib(n - 1) + fib(n - 2)

def main():
    iterations = 1000
    start = time.perf_counter()
    
    result = 0
    for _ in range(iterations):
        result += fib(19)
        
    end = time.perf_counter()
    
    total_ms = (end - start) * 1000.0
    avg_us = (total_ms * 1000.0) / iterations
    
    print("--- Python Benchmark ---")
    print(f"Fibonacci(19) runs: {iterations}")
    print(f"Total time: {total_ms:.2f} ms")
    print(f"Time per execution: {total_ms / iterations:.2f} ms")
    print(f"Sanity check result sum: {result}")

if __name__ == "__main__":
    main()
